src/pages/Blog/BlogItem/BlogItem.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { FC } from "react";
+import { useLoaderData } from "react-router-dom";
+
+const BlogItem: FC = () => {
+  const data = useLoaderData();
+  if (!data) {
+    return <div>Loading...</div>;
+  }
+  const hasAuthor = (value: unknown): value is { author: string } =>
+    typeof value === "object" && value !== null && "author" in value;
+  if (!hasAuthor(data)) {
+    throw new Error("Invalid data");
+  }
+
+  const { author, ...post } = data;
+  if (!isPost(post)) {
+    throw new Error("Invalid post");
+  }
+  const { title, body, tags, reactions } = post;
+  return (
+    <div>
+      <h1>{title}</h1>
+      <p>{body}</p>
+      <p>Author: {author}</p>
+      <p>Tags: {tags.join(", ")}</p>
+      <p>Reactions: {reactions}</p>
+    </div>
+  );
+};
+
+export type PostProps = {
+  id: number;
+  title: string;
+  body: string;
+  userId: number;
+  tags: string[];
+  reactions: number;
+};
+export const isPost = (post: unknown): post is PostProps => {
```

Comment:
move this whole function somewhere, looks so awful xD

Reply:
yup it does

Reply:
implemented the same logic using 'zod' it is much cleaner now